src/Main.ts

```diff
@@ -1120,6 +1120,74 @@ export class Main {
         }
     }
 
+    public async getChannelInfo(opts: {
+        slack_channel_id: string,
+        slack_bot_token: string,
+        team_id: string,
+    }): Promise<ConversationsInfoResponse|void> {
+        let slackClient: WebClient | undefined;
+        let teamEntry: TeamEntry | null = null;
+        let teamId: string = opts.team_id;
+
+        const existingChannel = opts.slack_channel_id ? this.rooms.getBySlackChannelId(opts.slack_channel_id) : null;
+
+        if (existingChannel) {
+            throw Error("Channel is already bridged! Unbridge the channel first.");
+        }
+
+        if (opts.slack_bot_token) {
+            if (!opts.slack_bot_token.startsWith("xoxb-")) {
+                throw Error("Provided token is not a bot token. Ensure the token starts with xoxb-");
+            }
+            // We may have this team already and want to update the token, or this might be new.
+            // But first check that the token works.
+            try {
+                teamId = await this.clientFactory.upsertTeamByToken(opts.slack_bot_token);
+                log.info(`Found ${teamId} for token`);
+            } catch (ex) {
+                log.error("Failed to action link because the token couldn't used:", ex);
+                throw Error("Token did not work, unable to get team");
+            }
+        }
+
+        // else, assume we have a teamId
+        if (teamId) {
```

Comment:
We should always have a teamId. If the team doesn't exist yet, the calling function should make sure we do create it.

Reply:
The calling function now ensures it's a non-empty string.
Removed the condition, keeping its content.